src/core/annotation.js

```diff
@@ -1097,7 +1097,14 @@ class WidgetAnnotation extends Annotation {
   }
 
   async save(evaluator, task, annotationStorage) {
-    if (this.data.fieldValue === annotationStorage[this.data.id]) {
+    const value = annotationStorage[this.data.id];
+    if (
+      this.data.fieldValue === value ||
+      value === null ||
+      value === undefined
+    ) {
+      // value === null when a choice widget
+      // hasn't been changed
```

Comment:
Nit: It looks like this comment will fit on *one* line, and please remember to end comments with a period as well.

This applies to the comments below as well.